### ISSUE TITLE
build issue with UFCS not behaving consistently

### ISSUE THREAD

**Issue by lightness1024** [nim-lang/Nim#7461]
```
import options
template p[T](): untyped = none(T)
debugecho p[int]()
```
fails with:

> Error: type mismatch: got <int>
> but expected one of: 
> proc none(T: typedesc): Option[T]
> 
> expression: none(T)

but, switch to method style call:

```
import options
template p[T](): untyped = T.none
debugecho p[int]()
```
now we're good.

note that to fix the build it's also possible to use `none(typedesc[T])` but that does not respect the principle of least astonishment.

**Comment by ringabout** [comment]
Another workaround:
```nim
import options
template p[T](): untyped = none[T]()
debugecho p[int]()
```

**Comment by metagn** [comment]
fixed by #22535
